src/langauge-servers/langaugeServer.ts

```diff
@@ -0,0 +1,119 @@
+    const writer = new rpc.WebSocketMessageWriter(socket);
+    let repo: Repo = null;
+
+    const langserverProcessName = 'Rookout-' + startConfig.LangaugeName + '-LangServer'
+    const serverConnection = bridgeServer.createServerProcess(langserverProcessName, startConfig.langserverCommand, startConfig.langserverCommandArgs)
+    const socketConnection = bridgeServer.createConnection(reader, writer, () => {socket.dispose(); serverConnection.dispose()});
+    
+
+    bridgeServer.forward(socketConnection, serverConnection, message => {
+        if (rpc.isRequestMessage(message)) {
+            if (message.method === lsp.InitializeRequest.type.method) {
+                const initializeParams = message.params as lsp.InitializeParams;
+                initializeParams.processId = process.pid;
+                
+                const repoId = initializeParams.workspaceFolders[0].uri.replace('file:///', '')
+                repo = repStore.getRepoById(repoId)
+
+                // rootUri and rootPath are considered deprecated by the vscode's lsp and they are the only way to indicate 
+                // to the language server the workspace folder
+                initializeParams.rootUri = initializeParams.workspaceFolders[0].uri = initializeParams.workspaceFolders[0].name = 'file://' + repo.fullpath
+                initializeParams.rootPath = repo.fullpath
+            }
+
+            if (message.method === lsp.DefinitionRequest.type.method && message.id) {
+                definitionsIds.add(message.id)
+
+                message.params.textDocument.uri = message.params.textDocument.uri.replace('file://', 'file://' + repo.fullpath)
+            }
+
+            if (message.method === lsp.FoldingRangeRequest.type.method ||
+                message.method === lsp.CodeLensRequest.type.method) {
+                    message.params.textDocument.uri = message.params.textDocument.uri.replace('file://', 'file://' + repo.fullpath)
+                }
+        }
+
+        if (rpc.isNotificationMessage(message)) {
+            if (message.method === lsp.DidOpenTextDocumentNotification.type.method || 
+                message.method === lsp.DidCloseTextDocumentNotification.type.method) {
+                
+                    const fileRelativePath = message.params.textDocument.uri.replace('file://', '')
+                    message.params.textDocument.uri = 'file://' + repo.fullpath + fileRelativePath
+            }
+            
+            // The frontend might send didChange requests which are wrong because of monaco-in-react behavior, so we ignore this kind of request
+            if (message.method === lsp.DidChangeTextDocumentNotification.type.method) {
+                message.method = "rookout-dummy"
+            }
+        }
+
+        if (rpc.isResponseMessage(message)) {
+            if (definitionsIds.has(message.id)) {
+                definitionsIds.delete(message.id)
+                
```

Comment:
why do we do this?

Reply:
In order to run `fixDefinitionResultsPath` we need to know which response message is "go-to-definition" response message, so we save up each id of go-to-definition request.
It is explained above the Set's declaration.